lib/Perl/Critic/Policy/Subroutines/ProhibitUnusedPrivateSubroutines.pm

```diff
@@ -75,6 +82,18 @@ sub _parse_private_name_regex {
 sub violates {
     my ( $self, $elem, $document ) = @_;
 
+    # if we're skipping, return right away
+    $self->{skipping} and return;
+
+    # A "use" not a sub declaration?
+    if ($elem->isa('PPI::Statement::Include')) {
+        # see if we're using something that indicates we shoud skip from now on
+        if ($self->{_skip_when_using}{ $elem->module }) {
+            $self->{skipping} = 1;
+        }
+        return;
+    }
+
```

Comment:
Let's see if we can generalize (and hide) some of that logic. So it might look like:

```
my @skip_modules = keys %{ $self->_skip_when_using }; 
return if any { $document->is_using($_) } @skip_modules;
```

If you really must optimize and cache the result:

```
my @skip_modules = keys %{ $self->_skip_when_using }; 
return if $self->{skipping} ||= any { $document->is_using($_) } @skip_modules;
```

And then `Perl::Critic::Document::is_using()` would implement the logic to scan the document for `use` statements. You could even go further and create a convenience interface like this:

```
my @skip_modules = keys %{ $self->_skip_when_using }; 
return if $self->{skipping} ||= $document->is_using(any => \@skip_modules);
```